### PR TITLE
Feat: Set NODE_ENV to 'test' like Jest CLI does

### LINKED ISSUE #50: jest and NODE_ENV environment variable
When I run jest my process.env.NODE_ENV is equal to `test`.
But when I run mutation test it is not equal to `test` (it's `undefined`)

### PR DESCRIPTION
**Pull request by Archcry**
Resolves #50.

Also did some boy scouting and bumped the mutation score to 100% :+1:

### CODE REVIEW

File: src/JestTestRunner.ts [hunk 1]
```diff
@@ -33,6 +42,14 @@ export default class JestTestRunner extends EventEmitter implements TestRunner {
     };
   }
 
+  private setNodeEnv() {
+    // Jest CLI will set process.env.NODE_ENV to 'test' when it's null, do the same here
+    // https://github.com/facebook/jest/blob/master/packages/jest-cli/bin/jest.js#L12-L14
+    if (this.processEnvRef.NODE_ENV == null) {
```
Comment (simondel) [inline src/JestTestRunner.ts]: Why not use a triple check?

File: test/unit/JestConfigEditorSpec.ts [hunk 2]
```diff
@@ -40,12 +45,20 @@ describe('JestConfigEditor', () => {
...
+  it('should call the ReactScriptsJestConfigLoader loadConfig method when no project is defined', () => {
     config.set({ jest: { project: 'react' } });
 
     jestConfigEditor.edit(config);
 
-    assert(reactConfigLoaderStub.loadConfig.calledOnce, 'ReactConfigLoader loadConfig not called');
+    assert(reactScriptsJestConfigLoaderStub.loadConfig.calledOnce, 'ReactScriptsJestConfigLoader loadConfig not called');
+  });
+
+  it('should call the ReactScriptsTSJestConfigLoader loadConfig method when no project is defined', () => {
```
Comment (simondel) [inline test/unit/JestConfigEditorSpec.ts]: But a project is defined in the test

File: src/JestTestRunner.ts (hunk 1 above)
Comment (unlight) [reply src/JestTestRunner.ts]: `undefined` also need to

Comment (unlight) [reply src/JestTestRunner.ts]: `jest-cli` does not use triple equal check, `undefined` need to be involved in comparison too

Comment (Archcry) [reply src/JestTestRunner.ts]: Ye, are we going the same route as jest or is `if(this.processEnvRef.NODE_ENV)` sufficient? I don't see a situation in which they're not equivalent.

File: test/unit/JestConfigEditorSpec.ts (hunk 2 above)
Comment (Archcry) [reply test/unit/JestConfigEditorSpec.ts]: Will update the tests, it seems their description is not correct atm.

**Pull request merged**